java/ql/test/library-tests/environment/SystemPropertyAccess.java

```diff
@@ -0,0 +1,25 @@
+import java.io.File;
+import java.util.Properties;
+import org.apache.commons.lang3.SystemUtils;
+
+public class SystemPropertyAccess {
+    private static final Properties SYSTEM_PROPERTIES = System.getProperties();
+
+    void test() {
+        System.getProperty("os.name");
+        System.getProperty("os.name", "default");
+        System.getProperties().getProperty("os.name");
+        System.getProperties().get("java.io.tmpdir");
+        SYSTEM_PROPERTIES.getProperty("java.home");
+        SYSTEM_PROPERTIES.get("file.encoding");
```

Comment:
@smowton is there a way to make these tests pass?

Reply:
The problem might be that you are using local flow, but `SYSTEM_PROPERTIES` is a field defined outside of the method, therefore local flow does not consider it (if I understand it correctly). Maybe making it a local variable works.

(I hope it is alright that I answered this)

Reply:
> (I hope it is alright that I answered this)

Yea, totally fine. Your feedback so far has been really helpful.

>  Maybe making it a local variable works.

It would, but I'd appreciate if something like `localDataFlow` supported static variable declarations like this

Reply:
It shouldn't really because this flow is not local. I think I answered this in a different forum already, but you'd need to make a predicate that incorporates both local flow and initializer -> read steps.